base-server/index.js

```diff
@@ -355,6 +369,45 @@ export class BaseServer {
     this.listenNotes = {}
     bindBackendProxy(this)
 
+    let end = (meta, queue, queueKey, ...args) => {
+      this.actionToQueue.delete(meta.id)
+      if (queue?.length() === 0) {
```

Comment:
Good question, it should never be missed at this point, but it was. 

The issue was that inside `on('processed')`, I determine whether an action belongs to the queue by using `actionToQueue` map. 
I save an action to this map by `meta.id`, but inside `on('processed')` I lookup the map by `meta.id` and `action.id` (because I may catch `logux/undo`, which has different `meta.id` but might have `action.id` of some action in the queue, in which case I need to undo all other actions in that queue). 

However, after processing, I always deleted an action from the `actionToQueue` by `meta.id`, so if it was `logux/undo` action, the following happened:

1. `on('processed')` got `logux/undo` with `action.id = '1'` and `meta.id = '2'`
2. `actionToQueue` has an action with `meta.id = '1'` equal to `action.id = '1'` so we process `logux/undo`
3. we undo all remaining actions in this queue and clear the queue
4. we try to delete this action (`logux/undo`) from the `actionToQueue` map by `meta.id = '2'` and delete nothing
5. the queue is deleted because it is empty (all actions are undone and removed from it)
6. `on('processed')` got some other action with `meta.id = '1'`
7. `actionToQueue` has an action with `meta.id = '1'` so we process it
8. we try to use the queue which was deleted in step `5`

Instead, in step `4` we should have deleted an action from `actionToQueue` using `action.id` so that in step `7` an action would be ignored because it was undone.